api/v1alpha1/reaper_types.go

```diff
@@ -106,6 +106,9 @@ type ReaperSpec struct {
 
 	// Tolerations applied to the Reaper pods
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Container-scoped SecurityContext
+	SecurityContext corev1.SecurityContext `json:"securityContext,omitempty"`
```

Comment:
I would change this to a pointer so we can more easily determine whether the property has been set. If its not set, then we go ahead and apply our default settings.

We probably want to add a SecurityContext for the `reaper-schema-init` init container as well.